app/controllers/api/internal/helper/purchases_controller.rb

```diff
@@ -526,13 +528,24 @@ def reassign_purchases
         end
       end
 
-      count += 1 if purchase.save
+      if purchase.save
+        count += 1
+        reassigned_purchase_ids << purchase.id
+      end
+    end
+
+    if reassigned_purchase_ids.any?
+      CustomerMailer.grouped_receipt(reassigned_purchase_ids).deliver_later(queue: "critical")
+    end
+
+    if target_user
+      Purchase.where(email: to_email, purchaser_id: nil).update_all(purchaser_id: target_user.id)
```

Comment:
Can we just remove the `&& purchase.purchaser_id.present?` on line 515 instead of this? Not sure why we'd want to check that, and would be nice to keep everything in the same update query

Reply:
We shouldn't need this now (and actually that conditional on line 515ish should probably just be replaced with `purchase.purchaser_id = target_user&.id`)